source/d3d9/runtime_d3d9.cpp

```diff
@@ -378,19 +484,149 @@ namespace reshade::d3d9
 
 		if (depthstencil != nullptr)
 		{
-			if (depthstencil == _depthstencil_replacement)
+			if (depthstencil == get_depthstencil_replacement())
 			{
+				// for the next tables, we need to get back to the original depthstencil ref
 				depthstencil = _depthstencil;
 			}
 
-			const auto it = _depth_source_table.find(depthstencil.get());
+			if (!_preserve_depth_buffer)
+			{
+				// original depth source table update
+				const auto it = _depth_source_table.find(depthstencil.get());
+
+				if (it != _depth_source_table.end())
+				{
+					it->second.drawcall_count = _drawcalls;
+					it->second.vertices_count += vertices;
+				}
+			}
+			else
+			{
+				_preserve_drawcalls++;
+				_preserve_vertices += vertices;
+
+				depthstencil = get_depthstencil_replacement();
+			}
+		}
+	}
+	void runtime_d3d9::before_clear(com_ptr<IDirect3DSurface9> depthstencil)
```

Comment:
Make the argument a `const com_ptr<IDirect3DSurface9> &depthstencil` to avoid some `AddRef` and `Release` calls.